app/controllers/users_controller.rb

```diff
@@ -1,6 +1,10 @@
 class UsersController < ApplicationController
-  skip_before_action :authenticate_user!, only: :show
-  before_action :set_user
+  before_action :set_user, only: %i[show edit update]
+
+  def index
+    users = policy_scope(User)
+    @users = params[:genre_ids] ? users.joins(:preferences).where(preferences: { genre_id: params[:genre_ids] }) : users
```

Comment:
Good idea to let them filter the search~
One question on the ternary here: where are the `params[:genre_ids]` coming from?

Reply:
we'll have a url like `tokyodnd.com/users?genre_ids=[1,2,3]` - the part after the question mark are params that we can access in the controller
the idea is that when a GM looks for players for their group (a button on their group's page) we'll automatically pass the genre id based on the rulebook they've chosen